### PR TITLE
Impute infection dates

### PR DESCRIPTION
**Pull request by phoenixAja**

### CODE REVIEW

**Comment by phoenixAja** [comment]
@lucymli how does `bin/03_impute_infection_dates.R` look to you? Did i miss anything important?

File: bin/03_impute_infection_dates.R [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+trs <- mapply(function (x, y) {
+  tr <- read.newick(x)
+  node_dates <- read_tsv(
+    y, comment="#", col_names=c(
+      "node", "date", "date_median", "date_lower", "date_upper"
+      )) %>% filter(as.character(date)!="--")
+  
+  tr <- keep.tip(tr, which(tr$tip.label %in% node_dates$node))
+  
+  tr$tip.label %<>% paste(node_dates$date[match(., node_dates$node)], sep="_")
```
Comment (lucymli) [inline bin/03_impute_infection_dates.R:25]: lines 18-25 should be updated to use the metadata.tsv table.
Suggested change:
```suggestion
  node_dates <- read_tsv(y)
  tr <- keep.tip(tr, which(tr$tip.label %in% node_dates$gisaid_epi_isl))
  tr$tip.label %<>% paste(node_dates$date[match(., node_dates$gisaid_epi_isl)], sep="_")
```

File: bin/03_impute_infection_dates.R [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+select_nodes <- c("california"="NODE_0000002",
+                  "france"="NODE_0000017",
+                  "guangdong"="NODE_0000012",
+                  "hongkong"="NODE_0000001",
+                  "hubei"="NODE_0000005",
+                  "italy"="NODE_0000008",
+                  "japan"="NODE_0000005",
+                  "shanghai"="NODE_0000001"
+)
```
Comment (lucymli) [inline bin/03_impute_infection_dates.R:45]: This step isn't necessary anymore. So lines 37-47 can be removed.
```suggestion
```

File: bin/03_impute_infection_dates.R (hunk 1 above)
Comment (phoenixAja) [reply bin/03_impute_infection_dates.R:25]: oh perfect, i totally forgot about our earlier conversation about `gisaid_epi_isl`

**Comment by phoenixAja** [comment]
great thanks @lucymli

**Pull request merged**